pkg/scheduler/factory/factory.go

```diff
@@ -488,6 +488,10 @@ func (c *configFactory) invalidatePredicatesForPv(pv *v1.PersistentVolume) {
 		invalidPredicates.Insert(predicates.MaxAzureDiskVolumeCountPred)
 	}
 
+	if pv.Spec.CSI != nil && utilfeature.DefaultFeatureGate.Enabled(features.AttachVolumeLimit) {
```

Comment:
I don't want to hold this PR, but please send a follow-up PR to add tests for these invalidation cases.

Reply:
okay sure.